### PR TITLE
Remove unused param & fix naming

### PR DESCRIPTION
**Pull request by QubitPi**
Address https://github.com/yahoo/fili/issues/273, @archolewa

### CODE REVIEW

File: fili-core/src/main/java/com/yahoo/bard/webservice/web/ratelimit/DefaultRateLimiter.java [hunk 1]
```diff
@@ -122,8 +122,8 @@ protected AtomicInteger getCount(String userName) {
      * @param initialCount  Initial count that we're incrementing and checking against the limit
      * @param requestLimit  Limit to check the incremented initial count against
      *
-     * @return True if the incremented count is <= the request limit, false if it's gone over and the request limit
```
Comment (QubitPi) [inline fili-core/src/main/java/com/yahoo/bard/webservice/web/ratelimit/DefaultRateLimiter.java]: `<=` doesn't pass checkstyle.

Comment (QubitPi) [inline fili-core/src/main/java/com/yahoo/bard/webservice/web/ratelimit/DefaultRateLimiter.java]: `<=` doesn't pass checkstyle

File: fili-core/src/main/resources/moduleConfig.properties [hunk 2]
```diff
@@ -78,8 +78,8 @@ bard__lucene_index_path = [SET ME IN APPLICATION CONFIG]
 # Lucene search timeout in milliseconds
 bard__lucene_search_timeout_ms = 600000
 
-# setting for maximum allowed results without any filters - used for /dim/values endpoint
-bard__max_results_without_filters = 10000
+# setting for maximum allowed results per page - used for /dim/values endpoint
+bard__max_results_per_page = 10000
```
Comment (cdeszaq) [inline fili-core/src/main/resources/moduleConfig.properties]: These config name changes need to be called out in the changelog I think.

File: fili-core/src/main/java/com/yahoo/bard/webservice/web/util/PaginationParameters.java [hunk 3]
```diff
@@ -25,17 +25,17 @@
     private static final String FIRST = "first";
     private static final String LAST = "last";
 
-    private static final int DEFAULT_MAX_RESULTS_WITHOUT_FILTERS = 10000;
-    private static final int MAX_RESULTS_WITHOUT_FILTER = SYSTEM_CONFIG.getIntProperty(
-            SYSTEM_CONFIG.getPackageVariableName("max_results_without_filters"),
-            DEFAULT_MAX_RESULTS_WITHOUT_FILTERS
+    private static final int DEFAULT_MAX_RESULTS_PER_PAGE = 10000;
+    private static final int MAX_RESULTS_PER_PAGE = SYSTEM_CONFIG.getIntProperty(
+            SYSTEM_CONFIG.getPackageVariableName("max_results_per_page"),
```
Comment (cdeszaq) [inline fili-core/src/main/java/com/yahoo/bard/webservice/web/util/PaginationParameters.java]: Changing this config param name is backwards incompatible. Can we continue to use the old param, logging a warning if it's still used, but respect it, while looking for the new one if it's there?

The order would be something like:

```groovy
MAX_RESULTS_PER_PAGE = max_results_without_filter 
    ? logErrorAndReturnMaxResultsWoFilterValue
    : max_results_per_page;
```

**Comment by michael-mclawhorn** [comment]
@QubitPi Either resolve @cdeszaq 's compatibility concerns or close, please

**Pull request closed without merging**